/*
 *              Copyright (c) 2016-2019, Nuralogix Corp.
 *                      All Rights reserved
 *
 *      THIS SOFTWARE IS LICENSED BY AND IS THE CONFIDENTIAL AND
 *      PROPRIETARY PROPERTY OF NURALOGIX CORP. IT IS
 *      PROTECTED UNDER THE COPYRIGHT LAWS OF THE USA, CANADA
 *      AND OTHER FOREIGN COUNTRIES. THIS SOFTWARE OR ANY
 *      PART THEREOF, SHALL NOT, WITHOUT THE PRIOR WRITTEN CONSENT
 *      OF NURALOGIX CORP, BE USED, COPIED, DISCLOSED,
 *      DECOMPILED, DISASSEMBLED, MODIFIED OR OTHERWISE TRANSFERRED
 *      EXCEPT IN ACCORDANCE WITH THE TERMS AND CONDITIONS OF A
 *      NURALOGIX CORP SOFTWARE LICENSE AGREEMENT.
 */
syntax = "proto3";

import "google/api/annotations.proto";
import "google/protobuf/timestamp.proto";

option cc_enable_arenas = true;
option java_package = "ai.nuralogix.dfx.auth.v1";
option java_outer_classname = "Auth";
option go_package = "bitbucket.org/nuralogix/dfx-protobuffs-go/dfx/auth/v1;auth";
option csharp_namespace = "Nuralogix.Dfx.Auth.V1";

package dfx.auth.v1;
 
service API {
  rpc ValidateToken (ValidateTokenRequest) returns (ValidateTokenResponse) {
    option (google.api.http) = {
      post: "/api/v1/auth/validate-token"
      body: "*"
    };
  }
  rpc Login (LoginRequest) returns (LoginResponse) {
    option (google.api.http) = {
      post: "/api/v1/auth/login"
      body: "*"
    };
  }
  rpc Logout (LogoutRequest) returns (LogoutResponse) {
    option (google.api.http) = {
      post: "/api/v1/auth/logout"
    };
   }
  rpc SwitchEffectiveOrganization (SwitchEffectiveOrganizationRequest) returns (SwitchEffectiveOrganizationResponse) {
    option (google.api.http) = {
      post: "/api/v1/auth/switch-effective-organization"
      body: "*"
    };
  }
 }

// ValidateToken
message ValidateTokenRequest {
  string token = 1;
  string license_id = 2;
  string organization_identifier = 3;
}
message ValidateTokenResponse {
  google.protobuf.Timestamp valid_until = 1;
}

// Login
message LoginRequest {
  string email = 1;
  string password = 2;
  string organization_identifier = 3;
}
message LoginResponse {
  string token = 1;
}

// Logout
message LogoutRequest {}
message LogoutResponse {}

// SwitchEffectiveOrganization
message SwitchEffectiveOrganizationRequest {
  string effective_organization_identifier = 1;
}
message SwitchEffectiveOrganizationResponse {
  string token = 1;
}